marketplace/plugins/documentdb/lib/index.ts

```diff
@@ -0,0 +1,207 @@
+import { QueryError, QueryResult, QueryService, ConnectionTestResult } from '@tooljet-marketplace/common';
+import { SourceOptions, QueryOptions } from './types';
+const { MongoClient } = require('mongodb');
+const JSON5 = require('json5');
+import { EJSON } from 'bson';
+
+const tlsCAFilePath = process.env.TLS_CA_FILE_PATH || 'global-bundle.pem';
```

Comment:
@Abhinav-CHD This is expected to infer from the deployment configuration  or a predefined directory. What we'd rather want is to have it inferred from the user when configuring the datasource.

Can we have the CA file content passed down from the queryOptions and then written onto a temporary directory to fit with the expected filepath behaviour?

Something like:
```js
 const tlsCAFile = await this.writeTempCACertificate(queryOptions.tlsCAFile, datasourceId)

.....
.....

 async writeCACertificate(caCertificate: string, dataSourceId: string): Promise<string> {
    const tempDir = await fs.mkdtemp(path.join(os.tmpdir(), 'ds-mongo-ca-'));
    const uniqueFilename = `${dataSourceId}.pem`;
    const caFilePath = path.join(tempDir, uniqueFilename);
    await fs.writeFile(caFilePath, caCertificate);
    return caFilePath;
}
```

Reply:
We need to use temporary directory as some deployment platform can be readonly, where writing to temp will only work.

Reply:
We can also store TLS file as in env/yaml file instead of doing that. It is given here in this [link](https://www.ibm.com/docs/en/mas-cd/continuous-delivery?topic=cawsd-configuring-maximo-application-suite-amazon-web-services-documentdb). Should I use this approach instead?

Reply:
@Abhinav-CHD, we cannot do that. This data source will be dynamically added. We cannot have specific authentication set up as a deployment dependency each time a new Amazon DocumentDB is added.